akka-stream/src/main/scala/akka/stream/scaladsl/Tcp.scala

```diff
@@ -70,6 +72,15 @@ object Tcp extends ExtensionId[Tcp] with ExtensionIdProvider {
   def lookup() = Tcp
 
   def createExtension(system: ExtendedActorSystem): Tcp = new Tcp(system)
+
+  // just wraps/unwraps the TLS byte events to provide ByteString, ByteString flows
+  private val tlsSupport: BidiFlow[ByteString, TLSProtocol.SendBytes, TLSProtocol.SslTlsInbound, ByteString, NotUsed] = BidiFlow.fromFlows(
+    Flow[ByteString].map(TLSProtocol.SendBytes),
+    Flow[TLSProtocol.SslTlsInbound].collect {
+      case sb: TLSProtocol.SessionBytes ⇒ sb.bytes
+      // ignore other kinds of inbounds (currently only Truncated)
```

Comment:
Would writing `case _ : Truncated => // ignore on purpose` be better and not cause warnings?

Reply:
With an empty byte string downstream then you mean?

Reply:
Yeah I wonder if that would be fine?

Reply:
It is a bit weirder for the downstream consumer, that there may be an empty byte string emitted, but I guess I can do that.

Reply:
I almost agreed. But then I thought that we don't really need it since there will be no warnings with `collect`.